src/DIRAC/FrameworkSystem/Service/TokenManagerHandler.py

```diff
@@ -1,57 +1,75 @@
-""" TokenManager service
+"""TokenManager service responsible for token management, namely storing, updating,
+requesting new tokens for DIRAC components that have the appropriate permissions.
 
-    .. literalinclude:: ../ConfigTemplate.cfg
-      :start-after: ##BEGIN TokenManager:
-      :end-before: ##END
-      :dedent: 2
-      :caption: TokenManager options
+.. literalinclude:: ../ConfigTemplate.cfg
+    :start-after: ##BEGIN TokenManager:
+    :end-before: ##END
+    :dedent: 2
+    :caption: TokenManager options
+
+The most common use of this service is to obtain user currents with certain scope to return to the user for its purposes,
+or to provide the DIRAC service to perform asynchronous tasks on behalf of the user.
+This is mainly about the :py:meth:`export_getToken` method.
+
+.. image:: /_static/Systems/FS/TokenManager_getToken.png
+    :alt: https://dirac.readthedocs.io/en/integration/_images/TokenManager_getToken.png
+
+The service and its client have a mechanism for caching the received tokens.
+This helps reduce the number of requests to both the service and the Identity Provider (IdP).
+
+If the client has a valid **access token** in the cache, it is used until it expires.
+After that you need to update. The client can update it independently if on the server where it is in ``dirac.cfg``
+``client_id`` and ``client_secret`` of the Identity Provider client are registered.
+
+Otherwise, the client makes an RPC call to the **TornadoManager** service.
+It in turn checks the cache and if the ``access token`` is already invalid tries to update it using a ``refresh token``.
+If the required tokens is not in the cache, then the ``refresh token`` from :py:class:`TokenDB <DIRAC.FrameworkSystem.DB.TokenDB.TokenDB>`
+is taken and the **exchange token** request to Identity Provider is made. The received tokens are cached.
```

Comment:
Maybe all this should be in the TokenManageClient.py, rather than on the service side.

Reply:
I didn't find any instructions on this, maybe you're right. But if so I think it is necessary to take it as a rule for the description of all services.
It is possible to use it at documentation building.